### PR TITLE
 Refactor tree classifier tests

### PR DESCRIPTION
**Pull request by KranthiGV**
Addresses #108 

This PR:
- We extract common code of various tree classifiers into a new method and use it across.
- No "functional" changes made to tests

I feel we could do more but I'm unsure whether or not to proceed further. Overall, the current state looks good enough.
There are some specific tests such as single node, wrong data type that are left untouched.

Validation:
- All tests pass locally.

### CODE REVIEW

File: tests/test_sklearn_decision_tree_converters.py [hunk 1]
```diff
@@ -33,56 +33,57 @@ def test_random_forest_implementation(self):
...
 
-    # Shifted classes
+    # Random forest shifted classes
     def test_random_forest_classifier_shifted_labels_converter(self):
-        self._run_random_forest_classifier_converter(3, labels_shift=2, extra_config={"tree_implementation": "gemm"})
-        self._run_random_forest_classifier_converter(3, labels_shift=2, extra_config={"tree_implementation": "tree_trav"})
-        self._run_random_forest_classifier_converter(3, labels_shift=2, extra_config={"tree_implementation": "perf_tree_trav"})
+        self._run_tree_classification_converter(RandomForestClassifier, 3, labels_shift=2, extra_config={"tree_implementation": "gemm"}, n_estimators=10)
+        self._run_tree_classification_converter(RandomForestClassifier, 3, labels_shift=2, extra_config={"tree_implementation": "tree_trav"}, n_estimators=10)
+        self._run_tree_classification_converter(RandomForestClassifier, 3, labels_shift=2, extra_config={"tree_implementation": "perf_tree_trav"}, n_estimators=10)
```
Comment (KranthiGV) [inline tests/test_sklearn_decision_tree_converters.py]: Should we separate these into 3 different tests?

Comment (interesaaat) [reply tests/test_sklearn_decision_tree_converters.py]: Yes please!

Comment (KranthiGV) [reply tests/test_sklearn_decision_tree_converters.py]: Doing it now. Do you feel anything else is off that we should address?

File: tests/test_sklearn_decision_tree_converters.py [hunk 2]
```diff
@@ -33,56 +33,63 @@ def test_random_forest_implementation(self):
                     str(type(list(pytorch_model.operator_map.values())[0])) == dt_implementation_map[extra_config_param]
                 )
 
-    def _run_random_forest_classifier_converter(self, num_classes, extra_config={}, labels_shift=0):
+    # Used for classification tests
+    def _run_tree_classification_converter(self, model_type, num_classes, extra_config={}, labels_shift=0, **kwargs):
         warnings.filterwarnings("ignore")
         for max_depth in [1, 3, 8, 10, 12, None]:
```
Comment (KranthiGV) [inline tests/test_sklearn_decision_tree_converters.py:39]: For my info: On what basis did we originally "tune"/identify these depth values?

File: tests/test_sklearn_decision_tree_converters.py [hunk 3]
```diff
@@ -146,39 +153,36 @@ def test_decision_tree_tree_trav_regressor_converter(self):
...
-            X = np.random.rand(1, 1)
-            X = np.array(X, dtype=np.float32)
-            y = np.random.randint(1, size=1)
-            model = RandomForestClassifier(n_estimators=1).fit(X, y)
-            pytorch_model = hummingbird.ml.convert(model, "pytorch", extra_config={"tree_implementation": extra_config_param})
-            self.assertTrue(pytorch_model is not None)
-            np.testing.assert_allclose(model.predict_proba(X), pytorch_model.predict_proba(X), rtol=1e-06, atol=1e-06)
+    # Small tree perf_tree_trav implementation
+    def test_random_forest_perf_tree_trav_classifier_single_node_tree_converter(self):
+        self._run_random_forest_classifier_single_node_tree_converter(extra_config={"tree_implementation": "perf_tree_trav"})
```
Comment (KranthiGV) [inline tests/test_sklearn_decision_tree_converters.py:185]: Will restore if there's some reasoning I missed.

File: tests/test_sklearn_decision_tree_converters.py (hunk 2 above)
Comment (interesaaat) [reply tests/test_sklearn_decision_tree_converters.py:39]: We use these values based on the pipelines we used in the experiments in the paper.

**Comment by interesaaat** [comment]
Thanks for the refactoring @KranthiGV. This looks good!

File: tests/test_sklearn_decision_tree_converters.py (hunk 2 above)
Comment (KranthiGV) [reply tests/test_sklearn_decision_tree_converters.py:39]: I see. Will read it and reach out if there's anything. Thanks!

**Pull request merged**